connection_go18.go

```diff
@@ -160,6 +160,10 @@ func (mc *mysqlConn) watchCancel(ctx context.Context) error {
 	select {
 	default:
 	case <-ctx.Done():
+		killErr := mc.killSelfQuery()
+		if killErr != nil {
+			return killErr
```

Comment:
I am not sure, that killErr should override original error of context.
if kill query fails, the error will be written to log only.

Reply:
Added error composition